### ISSUE TITLE
Carousel: Accessibility - when image set to  'Attachment page', clicking and tabbing should result in same behavior

### ISSUE THREAD

**Issue by coder-karen** [Automattic/jetpack#39926]
### Issue

An image set to link to an attachment page opens a lightbox if Carousel is enabled, if using a keyboard and pressing enter. However if the image is clicked on with a mouse, it opens the attachment page (it seems there should be consistency here). If Carousel is not enabled, both clicking on it and tabbing to it then clicking / entering open the attachment page (this appears to be expected).

### How to reproduce

This was tested on self-hosted and WoA.

1. On a test site, ensure the full-screen Carousel option is enabled under Jetpack > Settings > Writing: `/wp-admin/admin.php?page=jetpack#/writing`
2. Create a new post and add an image block.
3. Ensure 'Link to attachment page' is set for the 'Link' option:

![Image](https://github.com/user-attachments/assets/92232050-3950-4e58-aa23-7655ea5376c2)

4. Save and preview
5. Click on the image - the attachment page will open
7. Then, tab onto the image and press enter. A lightbox will open instead.

### What should happen

I'd expect when clicking on the image with Carousel enabled, that a lightbox would open (or at the least, that the behavior between clicking and tabbing would be the same).

### Related conversations

https://github.com/Automattic/jetpack/pull/39913#pullrequestreview-2398585938
